Model/Product/ReplicaManager.php

```diff
@@ -143,21 +169,20 @@ function($sort) use ($mode) {
      * In order to avoid interfering with replicas configured directly in the Algolia dashboard,
      * we must know which replica indices are Magento managed and which are not.
      *
-     * @param string $indexName
+     * @param string $primaryIndexName
      * @param int $storeId
      * @param bool $refreshCache
      * @return array
      * @throws LocalizedException
      * @throws NoSuchEntityException
      */
-    protected function getPossibleMagentoReplicaSettings(string $indexName, int $storeId, bool $refreshCache = false): array
+    protected function getPossibleMagentoReplicaSettingsFromConfig(string $primaryIndexName, int $storeId, bool $refreshCache = false): array
```

Comment:
I'm not a huge fan of the "possible" wording, it brings some uncertainty whereas this should be clear and absolute, if the index name has the right prefix, it belongs to Magento, and we shouldn't care about it if it doesn't. 
(But maybe I don't have all the keys to understand here)

Reply:
Good call out. You know I agree with you.I mean it is conceivable that someone could create a replica directly in Algolia (outside of Magento) that uses the exact same prefixing approach that the integration does (which would fool the Magento extension into thinking it was its own managed replica) but that would be a very silly thing to do. I don't think we need to worry about that so I will rename this.